src/extension/importer/core/src/main/java/org/geoserver/importer/job/JobQueue.java

```diff
@@ -60,8 +71,19 @@ protected void afterExecute(Runnable r, Throwable t) {
             public void run() {
                 List<Long> toremove = new ArrayList<Long>();
                 for (Map.Entry<Long, Task<?>> e : jobs.entrySet()) {
-                    if (e.getValue().isCancelled() || (e.getValue().isDone() && e.getValue().isRecieved())) {
-                        toremove.add(e.getKey());
+                    if (e.getValue().isCancelled() || (e.getValue().isDone())) {
+                        try {
+                            ImportContext context = (ImportContext) e.getValue().get();
+                            if(context.delete()) {
```

Comment:
Not sure this is "valid", what if the task is done but the context is still running (there can be imports with many tasks, hundreds or thousands)?